datadog_checks_base/datadog_checks/base/utils/db/sql.py

```diff
@@ -24,3 +34,51 @@ def compute_exec_plan_signature(normalized_json_plan):
         return None
     with_sorted_keys = json.dumps(json.loads(normalized_json_plan), sort_keys=True)
     return format(mmh3.hash64(with_sorted_keys, signed=False)[0], 'x')
+
+
+class eventEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, decimal.Decimal):
+            return float(o)
+        return super(eventEncoder, self).default(o)
+
+
+def chunks(items, n):
+    for i in range(0, len(items), n):
+        yield items[i:i + n]
+
+
+def submit_exec_plan_events(events, tags, source):
+    """
+    Submit the execution plan events to the event intake
+    https://docs.datadoghq.com/api/v1/logs/#send-logs
+    """
+    ddtags = ','.join(tags)
+    hostname = datadog_agent.get_hostname()
+    service = next((t for t in tags if t.startswith('service:')), 'service:{}'.format(source))[len('service:'):]
+    timestamp = time.time() * 1000
+
+    def _to_log_event(e):
+        return {
+            'ddtags': ddtags,
+            'hostname': hostname,
+            'message': json.dumps(e, cls=eventEncoder),
+            'service': service,
+            'ddsource': source,
+            'timestamp': timestamp
+        }
+
+    for chunk in chunks(events, 100):
+        try:
+            r = requests.post(f"https://http-intake.logs.datadoghq.com/v1/input",
```

Comment:
nit: make a constant

Reply:
nit: I think it would be better to use a session that persists for connection pooling. I _think_ requests might do that behind the scenes but don't think that would work unless it's attached to the instance class. That can be a follow up though.

Reply:
good point, updated